storage/server/jsonrpc.go

```diff
@@ -4,34 +4,106 @@
 package main
 
 import (
-	"io/ioutil"
+	"bufio"
+	"io"
+	"flag"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"log"
 	"net"
+	"sync/atomic"
 )
 
-func spdkCommunicate(buf []byte) ([]byte, error) {
-	// TODO: use rpc_sock variable
-	conn, err := net.Dial("unix", *rpc_sock)
+var (
+	ErrJSONNoSpaceLeft  = errors.New("json: No space left")
+	ErrJSONNoSuchDevice = errors.New("json: No such device")
+
+	rpcID      int32 // json request message ID, auto incremented
+	rpc_sock = flag.String("rpc_sock", "/var/tmp/spdk.sock", "Path to SPDK JSON RPC socket")
+)
+
+// low level rpc request/response handling
+func call(method string, args, result interface{}) error {
```

Comment:
The most beautifully named function in all of OPI. 🤣

Reply:
lol